compiler/rustc_middle/src/ty/diagnostics.rs

```diff
@@ -355,6 +355,10 @@ pub fn suggest_constraining_type_params<'a>(
         ));
     }
 
+    // remove the suggestions that are not suitable for diagnostics
+    suggestions =
+        suggestions.into_iter().filter(|(span, _, _)| !span.from_expansion()).collect::<Vec<_>>();
```

Comment:
Then, could you use `in_derive_expansion` instead and leave a FIXME comment?

```suggestion
    // remove the suggestions that are not suitable for diagnostics
    suggestions =
        suggestions.into_iter().filter(|(span, _, _)| !span.in_derive_expansion()).collect::<Vec<_>>();
```

`from_expansion` skips suggestions for all kinds of macro including bang macros.